src/services/TodoService.ts

```diff
@@ -0,0 +1,158 @@
+import { useEffect } from 'react';
+import { addTodos, delateTodos, patchTodos } from '../api/todos';
+import { useTodoContext } from '../utils/hooks/useTodoContext';
+import { ErrorMessages } from '../types/ErrorMessages/ErrorMessages';
+import { Todo } from '../types/Todo/Todo';
+
+export const TodoService = () => {
+  const {
+    todos,
+    setTodos,
+    setErrorMessage,
+    showError,
+    setLoading,
+    setFocusInput,
+    inputRef,
+    focusInput,
+  } = useTodoContext();
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [focusInput, inputRef]);
+
+  const editTodo = async (id: number, data: Partial<Todo>) => {
+    try {
+      const editTodoItem = await patchTodos(id, data);
+
+      setTodos(currToto =>
+        currToto.map(todo => {
+          if (todo.id === id) {
+            return editTodoItem;
+          }
+
+          return todo;
+        }),
+      );
+    } catch {
+      showError(ErrorMessages.Edit);
+    }
+  };
+
+  const createTodo = async ({ userId, completed, title }: Todo) => {
+    try {
+      const newTodo = await addTodos({ userId, completed, title });
+
+      setTodos(curenTodos => [...curenTodos, newTodo]);
+    } catch (error) {
+      showError(ErrorMessages.Add);
+      throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)

Fix everywhere